ethereum/forge-test/GovernanceStructs.t.sol

```diff
@@ -0,0 +1,320 @@
+        bytes32 module,
+        uint8 action,
+        uint16 chain,
+        uint256 messageFee
+    ) public {
+        vm.assume(action != 3);
+
+        bytes memory encodedSetMessageFee = abi.encodePacked(
+            module,
+            action,
+            chain,
+            messageFee
+        );
+
+        assertEq(encodedSetMessageFee.length, 67);
+
+        vm.expectRevert("invalid SetMessageFee");
+
+        gs.parseSetMessageFee(encodedSetMessageFee);
+    }
+
+    function testParseSetMessageFeeWrongAction_KEVM(
+        bytes32 module,
+        uint8 action,
+        uint16 chain,
+        uint256 messageFee
+    ) public symbolic(address(gs)) {
+        testParseSetMessageFeeWrongAction(module, action, chain, messageFee);
+    }
+
+    // Needs loop invariant for unbounded bytes type
+    function testSetMessageFeeInvalidSize(bytes memory encodedSetMessageFee)
+        public
+    {
+        vm.assume(encodedSetMessageFee.length != 67);
+
+        vm.expectRevert();
+
+        gs.parseSetMessageFee(encodedSetMessageFee);
+    }
+
+    function testParseTransferFees(
+        bytes32 module,
+        uint16 chain,
+        uint256 amount,
+        bytes32 recipient
+    ) public {
+        uint8 action = 4;
+
+        bytes memory encodedTransferFees = abi.encodePacked(
+            module,
+            action,
+            chain,
+            amount,
+            recipient
+        );
+
+        assertEq(encodedTransferFees.length, 99);
+
+        GovernanceStructs.TransferFees memory tf =
+            gs.parseTransferFees(encodedTransferFees);
+
+        assertEq(tf.module, module);
+        assertEq(tf.action, action);
+        assertEq(tf.chain, chain);
+        assertEq(tf.amount, amount);
+        assertEq(tf.recipient, recipient);
+    }
+
+    function testParseTransferFees_KEVM(
+        bytes32 module,
+        uint16 chain,
+        uint256 amount,
+        bytes32 recipient
+    ) public symbolic(address(gs)) {
+        testParseTransferFees(module, chain, amount, recipient);
+    }
+
+    function testParseTransferFeesWrongAction(
+        bytes32 module,
+        uint8 action,
+        uint16 chain,
+        uint256 amount,
+        bytes32 recipient
+    ) public {
+        vm.assume(action != 4);
+
+        bytes memory encodedTransferFees = abi.encodePacked(
+            module,
+            action,
+            chain,
+            amount,
+            recipient
+        );
+
+        assertEq(encodedTransferFees.length, 99);
+
+        vm.expectRevert("invalid TransferFees");
+
+        gs.parseTransferFees(encodedTransferFees);
+    }
+
+    function testParseTransferFeesWrongAction_KEVM(
+        bytes32 module,
+        uint8 action,
+        uint16 chain,
+        uint256 amount,
+        bytes32 recipient
+    ) public symbolic(address(gs)) {
+        testParseTransferFeesWrongAction(module, action, chain, amount, recipient);
+    }
+
+    // Needs loop invariant for unbounded bytes type
```

Comment:
The main goal of this engagement was to formally verify Wormhole functionality, by using our tool KEVM to symbolically execute property tests. But we decided to limit the scope of this engagement to those tests that didn't include loops, and leave tests that did include loops to a follow-up engagement. This is because symbolically executing loops requires setting up a loop invariant, which requires more work than straight-line code. The `...InvalidSize` tests actually have a hidden loop because the parameter of type `bytes` (of arbitrary length) needs to be copied into memory. So they were left out of the present scope because they would need a loop invariant to be formally verified.